src/VueStorefront/PageResult/Navigation/NavigationPageResultHydrator.php

```diff
@@ -31,14 +39,104 @@ public function hydrate(PageLoaderContext $pageLoaderContext, CategoryEntity $ca
     {
         $this->pageResult->setCmsPage($cmsPageEntity);
 
-        $this->setBreadcrumbs($category, $pageLoaderContext->getContext()->getSalesChannel()->getNavigationCategoryId());
+        $this->setBreadcrumbs(
+            $category,
+            $pageLoaderContext->getContext()->getSalesChannel()->getNavigationCategoryId()
+        );
 
         $this->pageResult->setResourceType($pageLoaderContext->getResourceType());
         $this->pageResult->setResourceIdentifier($pageLoaderContext->getResourceIdentifier());
 
+        $this->getAvailableFilters();
+
         return $this->pageResult;
     }
 
+    /**
+     * @TODO: Destruct method into dedicated services and handlers
+     */
+    private function getAvailableFilters(): array
+    {
+        /**
+         * Get all listing slots
+         */
+        $listingSlots = $this->pageResult->getCmsPage()->getSections()->getBlocks()->getSlots()->filter(function($slot) {
+            /** @var $slot CmsSlotEntity */
+            return $slot->getType() === 'product-listing';
+        })->getElements();
+
+        /**
+         * Get the listing results of these slots
+         */
+        /** @var ProductListingResult[] $listings */
+        $listings = array_map(function($slot) {
+            /** @var $slot CmsSlotEntity */
+            return $slot->getData()->getListing();
+        }, $listingSlots);
+
+        /**
+         * For every listing create config from aggre
+         */
+        foreach($listings as $listing)
+        {
+            $sortings = $listing->getSortings();
+            $filters = $listing->getAggregations()->getElements();
+
+            $filters = array_map(function($aggregation) {
+                if($aggregation instanceof EntityResult) {
+                    return [
+                        'type' => 'term',
+                        'values' => array_map(function($element) {
+                            return $element->getName(); // Bad, because not all entities have names
+                        }, $aggregation->getEntities()->getElements())
+                    ];
+                }
+
+                if($aggregation instanceof StatsResult) {
+                    return [
+                        'type' => 'range',
+                        'values' => [
+                            'max' => $aggregation->getMax(),
+                            'min' => $aggregation->getMin()
+                        ]
+                    ];
+                }
+                if($aggregation instanceof TermsResult) {
+                    return [
+                        'type' => 'term',
+                        'values' => array_map(function($bucket) {
+                            /** @var $bucket Bucket */
+                            return $bucket->getKey();
+                        }, $aggregation->getBuckets())
+                    ];
+                }
+                if($aggregation instanceof MaxResult) {
```

Comment:
Why do you have to convert things? The requested data is known. Means: the client knows that the "price" aggregation must be a slider and returns min-max.